### PR TITLE
WFCORE-6259 Refactor SubsystemResourceDescriptionResolver implementation from wildfly-clustering-common to wildfly-controller

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.redhat.com/browse/WFCORE-6259

### CODE REVIEW

File: controller/pom.xml
Comment (yersan) [inline controller/pom.xml:154]: We are planning to move all test dependencies to the testbom. I think it reasonable to merge this as it is. 

@parsharma We will need to take into account this new test dependency and move it to the testbom. Sadly your PR is still in the queue since we need to get the WildFly Full counterpart merged before merging yours.

**Review by yersan — APPROVED** [review]

**Comment by yersan** [comment]
Would you mind adding the Jira number to each commit message?, we don't need to wait for a full CI, since it already passed.

**Comment by pferraro** [comment]
> Would you mind adding the Jira number to each commit message?, we don't need to wait for a full CI, since it already passed.

Done.

**Pull request merged**
